File: typescript-examples/browser-sdk-showcase/api/sanitize-html.ts
// https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/sanitizeHtml
import { BrowserContext, Page } from "playwright";
import { sanitizeHtml } from "@intuned/browser";

interface Params {
  removeStyles?: boolean;
  maxAttributeLength?: number;
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  await page.goto("https://books.toscrape.com");
  const removeStyles = params.removeStyles ?? true;
  const maxAttributeLength = params.maxAttributeLength ?? 100;
  const firstRow = page.locator("ol.row").locator("li").first();
  const html = await firstRow.innerHTML();
  const sanitizedHtml = sanitizeHtml({ html, removeStyles, maxAttributeLength });
  console.log("Sanitized HTML:");
  console.log(sanitizedHtml);
  return {
    sanitized_html: sanitizedHtml,
  };
}

File: python-examples/browser-sdk-showcase/api/sanitize-html.py
# https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/sanitize_html
from typing import TypedDict

from intuned_browser import sanitize_html
from playwright.async_api import Page

# from intuned_runtime import extend_payload


class Params(TypedDict):
    remove_styles: bool | None
    max_attribute_length: int | None


async def automation(page: Page, params: Params | None = None, **_kwargs):
    await page.goto("https://books.toscrape.com")
    remove_styles = params.get("remove_styles", True)
    max_attribute_length = params.get("max_attribute_length", 100)
    first_row = page.locator("ol.row").locator("li").first
    html = await first_row.inner_html()
    sanitized_html = sanitize_html(html, remove_styles=remove_styles, max_attribute_length=max_attribute_length)
    print("Sanitized HTML:")
    print(sanitized_html)
    return {
        "sanitized_html": sanitized_html,
    }
